### PR TITLE
.status.loadBalancer.ingress[0].ip is empty in AWS

### LINKED ISSUE #237: stable/jenkins - instructions don't work in AWS
`.status.loadBalancer.ingress[0].ip}` has no value in AWS since the ELB only returns the hostname.

### PR DESCRIPTION
**Pull request by ipedrazas**
Fixes #237

`.status.loadBalancer.ingress[0].ip}` has no value in AWS since the ELB only returns the hostname. I've added instructions to get the jenkins url using the hostname of the eLB instead of the IP.

### CODE REVIEW

**Comment by ipedrazas** [comment]
I signed it!

**Comment by lachie83** [comment]
Thanks @ipedrazas. How about we update to this?

```
kubectl describe svc {{ template "fullname" . }} --namespace {{ .Release.Namespace }} | grep Ingress | awk '{print $3}'
```

**Comment by ipedrazas** [comment]
Not sure about this.

Have you thought of the windows users? I'm not sure if Windows has `grep` and `awk`, which it might be a problem for some users... but I've just realised that this only makes sense if Helm supports Windows, which is not the case, yet (correct me if I'm wrong).

Another reason would be that my form nudges users to learn the harsh world of kubernetes schemas.

But my point was to provide a copy and paste instruction for users who are not familiar with kubernetes and helm.

**Comment by lachie83** [comment]
All fair points. I would be nice if we could determine what platform the cluster is running on to switch that to a conditional. It's just a bunch of information to go through to get started

**Comment by prydonius** [comment]
See https://github.com/kubernetes/charts/issues/84 for other alternatives.

```
kubectl get svc {{ template "fullname" . }} --template "{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"
```
would also work and doesn't require awk/grep.

**Comment by ipedrazas** [comment]
Happy to update it so we can have some consistency across charts 😄

**Comment by lachie83** [comment]
Let's go with what @prydonius suggested.

File: stable/jenkins/templates/NOTES.txt [hunk 1]
```diff
@@ -10,8 +10,14 @@
 {{- else if contains "LoadBalancer" .Values.Master.ServiceType }}
 **** NOTE: It may take a few minutes for the LoadBalancer IP to be available.                      ****
 ****       You can watch the status of by running 'kubectl get svc -w {{ template "fullname" . }}' ****
-  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  export SERVICE_IP=$(kubectl get svc {{ template "fullname" . }} --template "{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}")
+  
   echo http://$SERVICE_IP:{{ .Values.Master.ServicePort }}/login
+  
+If you're running Kubernetes in AWS, the ingress has not ip, but you can use the Hostname
```
Comment (lachie83) [inline stable/jenkins/templates/NOTES.txt]: You can yank this block now right?

Comment (ipedrazas) [reply stable/jenkins/templates/NOTES.txt]: yep, I've just realised. Editing this with a phone is not ideal, sorry.

File: stable/jenkins/templates/NOTES.txt [hunk 2]
```diff
@@ -10,8 +10,9 @@
 {{- else if contains "LoadBalancer" .Values.Master.ServiceType }}
 **** NOTE: It may take a few minutes for the LoadBalancer IP to be available.                      ****
 ****       You can watch the status of by running 'kubectl get svc -w {{ template "fullname" . }}' ****
-  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  export SERVICE_IP=$(kubectl get svc {{ template "fullname" . }} --template "{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}")
```
Comment (lachie83) [inline stable/jenkins/templates/NOTES.txt]: Can you please add ` --namespace {{ .Release.Namespace }}` to this?

**Comment by ipedrazas** [comment]
Done. Squashed my mess also.

**Comment by lachie83** [comment]
Looks like it's not escaped properly
```
helm install . --namespace=patch
Error: render error in "jenkins/templates/NOTES.txt": template: jenkins/templates/NOTES.txt:13:138: executing "jenkins/templates/NOTES.txt" at <.status.loadBalancer...>: can't evaluate field loadBalancer in type interface {}
```

**Comment by lachie83** [comment]
Here's an example https://github.com/kubernetes/charts/blob/7e971635642494b607118aab56e17a21508bcc7f/stable/ghost/templates/NOTES.txt#L21

**Comment by lachie83** [comment]
Perfect! Thanks @ipedrazas!

**Comment by lachie83** [comment]
LGTM

**Pull request merged**